### PR TITLE
Controller

### PR DESCRIPTION
**Pull request by lukman155**
Use controllers to handle requests and render empty views.
Understand Rails naming conventions.
Use params from browser requests in a safe way.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
+  get '/users/:user_id/posts(.:format)', to: 'posts#index'
+  get '/users/:user_id/posts/:id(.:format)', to: 'posts#show'
+  get '/users(.:format)', to: 'users#index'
+  get '/users/:id(.:format)', to: 'users#show'
```
Comment (youmari) [inline config/routes.rb]: - [ ] Nice adding the routes but it would be better to use the `resources` keyword it would shortcut a lot

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @lukman155 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: config/routes.rb (hunk 1 above)
Comment (lukman155) [reply config/routes.rb]: Thank you, I have implemented it

**Review by BregornOriginal — COMMENTED** [review]
Hi @lukman155 

Your project is complete!  There is nothing else to say other than... it's time to merge it 

Congratulations! 

giphy (7)

## Highlights

✅ Linters are passing
✅ Good commit history
✅ Good work implementing `resources` on your routes 💯 

## Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.
Cheers and Happy coding!:aplauso::aplauso::aplauso:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Please, remember to tag me in your question so I can receive the notification.
------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)[.](https://airtable.com/shrQAqnBwek5a0O0s)._)

**Review by BregornOriginal — COMMENTED** [review]
.

**Pull request merged**